ASM-5840 - [Noble] Memory leak in Torquebox

mdcondon opened the pull request:
Ready for review/merge

ripienaar commented:
I'm not the best person to review this - I am just curious what the `[Noble]` in the commit message means?

gavin-scott commented:
Noble was the customer that saw this initially and was included on the jira title.

ripienaar commented:
oh, right :)

gavin-scott commented on scripts/rpm/postInstall.sh:
```diff
@@ -14,6 +14,13 @@ touch /etc/carbon/storage-aggregation.conf
+# Update Gemfile.lock for Torquebox 3.0.2
+GEMFILE_LOCK=/opt/asm-deployer/Gemfile.lock
+ORIG_PERMS=`stat --printf=%a $GEMFILE_LOCK`
+chmod 755 $GEMFILE_LOCK
```
Why did you have to monkey with the permissions?

gavin-scott commented on scripts/rpm/postInstall.sh:
```diff
@@ -14,6 +14,13 @@ touch /etc/carbon/storage-aggregation.conf
+GEMFILE_LOCK=/opt/asm-deployer/Gemfile.lock
+ORIG_PERMS=`stat --printf=%a $GEMFILE_LOCK`
+chmod 755 $GEMFILE_LOCK
+sed -i -e 's/\(torque.*\)\((= 3.0.[0-9])\)/\1(= 3.0.2)/' $GEMFILE_LOCK
```
Probably be good to add a check that `$GEMFILE_LOCK` exists before we do this. Since it's not included in the rpm it's not guaranteed to exist.

gavin-scott replied:
Also realized while talking to you that we need to update the torquebox version in the asm-deployer Gemfile

mdcondon commented:
Right on all counts. Hopefully fixed.

Fixed asm-deployer-rpm postInstall.sh - add existence check for Gemfile.lock and removed perms operations which were not needed.

Updated asm-deployer Gemfile to specify TB 3.0.2 and submitted a PR [1400](https://github.com/dell-asm/asm-deployer/pull/1400)

Edited comment in defect to add PR for asm-deployer
